petab/v2/problem.py

```diff
@@ -125,13 +126,37 @@ def from_yaml(yaml_config: dict | Path | str) -> Problem:
         if isinstance(yaml_config, Path):
             yaml_config = str(yaml_config)
 
-        get_path = lambda filename: filename  # noqa: E731
         if isinstance(yaml_config, str):
-            path_prefix = get_path_prefix(yaml_config)
+            yaml_file = yaml_config
+            path_prefix = get_path_prefix(yaml_file)
             yaml_config = yaml.load_yaml(yaml_config)
             get_path = lambda filename: f"{path_prefix}/{filename}"  # noqa: E731
+        else:
+            yaml_file = None
+            get_path = lambda filename: filename  # noqa: E731
 
         if yaml_config[FORMAT_VERSION] not in {"2.0.0"}:
+            # If we got a path to a v1 yaml file, try to auto-upgrade
+            from tempfile import TemporaryDirectory
+
+            from ..versions import get_major_version
+            from .petab1to2 import petab1to2
+
+            if get_major_version(yaml_config) == 1 and yaml_file:
```

Comment:
This suggests that `get_major_version` should be used instead of `if str(yaml_config[petab.C.FORMAT_VERSION]) != "1":` in `petab1to2` as well.

Reply:
Correct.